foxtrot-core/src/main/java/com/flipkart/foxtrot/core/datastore/impl/hbase/HbaseRegions.java

```diff
@@ -0,0 +1,142 @@
+package com.flipkart.foxtrot.core.datastore.impl.hbase;
+
+import com.flipkart.foxtrot.common.hbase.HRegionData;
+import com.flipkart.foxtrot.common.util.CollectionUtils;
+import com.flipkart.foxtrot.core.exception.HbaseRegionExtractionException;
+import com.flipkart.foxtrot.core.exception.HbaseRegionMergeException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HRegionInfo;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.apache.hadoop.hbase.client.RegionLocator;
+import org.apache.hadoop.hbase.util.RegionSizeCalculator;
+
+import java.util.*;
+
+@Slf4j
+public class HbaseRegions {
+
+    private static final long BYTES_IN_GB = 1073741824;
```

Comment:
@santanusinha This is being done to merge smaller regions. Lots of regions become smaller in size over the time because records get deleted. These regions just stay there unless merged